### PR TITLE
Ps 5.7 myr 47

### PR DESCRIPTION
**Pull request by george-lorch**

### CODE REVIEW

**Comment by george-lorch** [comment]
https://jenkins.percona.com/view/5.7/job/mysql-5.7-param/1613/

File: mysql-test/suite/rocksdb/t/i_s_trx_rpl.test [hunk 1]
```diff
@@ -0,0 +1,42 @@
+--source include/have_rocksdb.inc
+--source include/master-slave.inc
+
+--disable_warnings
```
Comment (laurynas-biveinis) [inline mysql-test/suite/rocksdb/t/i_s_trx_rpl.test]: There is a rpl_skip_start_slave (sic?) arg for master-slave.inc instead

File: mysql-test/suite/rocksdb/t/i_s_trx_rpl.test [hunk 2]
```diff
@@ -0,0 +1,42 @@
...
+  eval insert into t1 values ($aa);
+  --inc $aa
+}
+--enable_query_log
+
+connection slave;
+show variables like 'rocksdb_rpl_skip_tx_api';
+--source include/start_slave.inc
+
+--let $it=0
```
Comment (laurynas-biveinis) [inline mysql-test/suite/rocksdb/t/i_s_trx_rpl.test]: Replace with wait_for_condition.inc, among other things this loop is spinning without any pause between iterations

File: storage/rocksdb/ha_rocksdb.cc [hunk 3]
```diff
@@ -4673,6 +4679,13 @@ bool ha_rocksdb::init_with_fields() {
   DBUG_RETURN(false); /* Ok */
 }
 
+bool ha_rocksdb::rpl_can_handle_stm_event() const {
+  if (rpl_skip_tx_api_var == TRUE && super_read_only == FALSE) {
```
Comment (laurynas-biveinis) [inline storage/rocksdb/ha_rocksdb.cc]: return !rpl_skip_tx_api_var || super_read_only

**Review by laurynas-biveinis — CHANGES_REQUESTED** [review]
Is there a TokuDB angle to this?

**Comment by george-lorch** [comment]
No TokuDB, PerconaFT does't have quite the same concept implemented, although it might be worth investigating if there is a way to use FT w/o transactions.

**Comment by george-lorch** [comment]
https://jenkins.percona.com/view/5.7/job/mysql-5.7-param/1633/

**Comment by george-lorch** [comment]
OK, changes made, jenkins looks good (except for the known instabilities)

File: storage/rocksdb/ha_rocksdb.cc [hunk 4]
```diff
@@ -4673,6 +4679,13 @@ bool ha_rocksdb::init_with_fields() {
   DBUG_RETURN(false); /* Ok */
 }
 
+bool ha_rocksdb::rpl_can_handle_stm_event() const {
+  if (rpl_skip_tx_api_var && !super_read_only) {
```
Comment (laurynas-biveinis) [inline storage/rocksdb/ha_rocksdb.cc]: if (foo) return false; return true; -> return !foo;

**Review by laurynas-biveinis — CHANGES_REQUESTED** [review]

**Comment by george-lorch** [comment]
trivial logic fix, new jenkins https://jenkins.percona.com/view/5.7/job/mysql-5.7-param/1649/

**Review by laurynas-biveinis — APPROVED** [review]

**Pull request merged**